compiler/rustc_session/src/config.rs

```diff
@@ -1296,6 +1296,29 @@ impl CrateType {
             CrateType::Executable | CrateType::Cdylib | CrateType::Staticlib => false,
         }
     }
+
+    pub fn shorthand(&self) -> &'static str {
+        match *self {
+            CrateType::Executable => "bin",
+            CrateType::Dylib => "dylib",
+            CrateType::Rlib => "rlib",
+            CrateType::Staticlib => "lib",
+            CrateType::Cdylib => "cdylib",
+            CrateType::ProcMacro => "proc-macro",
+        }
+    }
+
+    fn shorthands_display() -> String {
+        format!(
+            "`{}`, `{}`, `{}`, `{}`, `{}`, `{}`",
+            CrateType::Executable.shorthand(),
+            CrateType::Dylib.shorthand(),
+            CrateType::Rlib.shorthand(),
+            CrateType::Staticlib.shorthand(),
+            CrateType::Cdylib.shorthand(),
+            CrateType::ProcMacro.shorthand(),
```

Comment:
Suggestion: can you sort these alphabetically (for display) to make it easier for the user to follow? (I know the enum definition isn't sorted alphabetically).